aip/general/0121.md

```diff
@@ -53,6 +53,19 @@ database schema. In fact, having an API that is identical to the underlying
 database schema is actually an anti-pattern, as it tightly couples the surface
 to the underlying system.
 
+### Resource requirements
+
+For a valid resource-oriented API, the following must be satisfied:
+
+- Each resource **must** have a unique [resource name][] by which it can be
+  identified and referenced.
+- The resource API must support [Get][].
+- The resource API must support [List][], with the exception of [singleton
+  resources][].
```

Comment:
Why must it support List? Sometimes the datasets are large, so large that pagination becomes a complex problem or other business reasons may make it so that List is not imperative. I agree that Get should be required, but why List?

Reply:
This is from #993, @bgrant0607 requested this integration.

I think this was long the lines of normalizing as many operations as possible. 

If a list wasn't supported though, it'd be fairly problematic. For GCP,  how would GCloud do list? or Console show all of the resource?

Reply:
+1. We should really be driving towards more requirements and away from flexibility.

Anyway, from what I've seen, List is more commonly supported than Get currently.